test/metabase/api/action_test.clj

```diff
@@ -278,6 +281,13 @@
               (is (= false
                      (db/exists? Action :id action-id, :public_uuid (:public_uuid action-opts)))))))
 
+        (testing "Test that we cannot unshare an action if it's archived"
+          (let [action-opts (merge {:archived true} (shared-action-opts))]
```

Comment:
You're missing a test for the POST endpoint, not being able to share an action if it's archived